tcms/static/js/testrun_actions.js

```diff
@@ -192,6 +192,8 @@ Nitrate.TestRuns.Details.on_load = function() {
   });
   jQ('.status-link-button').bind('click', function() {
     showCaseRunsWithSelectedStatus(jQ('#id_filter')[0], jQ(this).data('param'));
```

Comment:
Thinking about this the entire current approach must be updated. 

This entire function is pointless because what it does is call fireEvent (which we want to remove) which submits a form to the current URL sending a bunch of GET parameters. From what I can see the only parameter which matters is status__pk (status_id should also work and avoid a JOIN) so you can add this directly in the href attribute.